op-batcher/batcher/driver.go

```diff
@@ -324,6 +331,66 @@ func (l *BatchSubmitter) loop() {
 	}
 }
 
+// checkRecentTxsOnStart Check to see if there was a batcher tx sent recently that
+// still needs more block confirmations before being considered finalized
+func (l *BatchSubmitter) checkRecentTxsOnStart() {
+	currBlock, err := l.Txmgr.BlockNumber(l.shutdownCtx)
+	if err != nil {
+		l.Log.Error("failed to retrieve current block number", "err", err)
+		return
+	}
+
+	currentBlock := new(big.Int).SetUint64(currBlock)
+	currentNonce, err := l.L1Client.NonceAt(l.shutdownCtx, l.Txmgr.From(), currentBlock)
+	if err != nil {
+		l.Log.Error("Failed to retrieve current nonce", "err", err)
+		return
+	}
+
+	rollupClient, err := l.EndpointProvider.RollupClient(l.shutdownCtx)
+	if err != nil {
+		l.Log.Error("Failed to get rollup client", "err", err)
+		return
+	}
+	rollupConfig, err := rollupClient.RollupConfig(l.shutdownCtx)
+	if err != nil {
+		l.Log.Error("Failed to retrieve rollup config", "err", err)
+		return
+	}
+	blockConfirms := rollupConfig.VerifierConfDepth
+
+	previousBlockBig := new(big.Int)
+	previousBlockBig.Sub(currentBlock, big.NewInt(int64(blockConfirms)))
+	previousNonce, err := l.L1Client.NonceAt(l.shutdownCtx, l.Txmgr.From(), previousBlockBig)
+	if err != nil {
+		l.Log.Error("Failed to retrieve previous nonce", "err", err)
+		return
+	}
+
+	if currentNonce == previousNonce {
+		l.Log.Info("No recent batcher txs detected. No need to wait for rollup sync")
+		return
+	}
+
+	l.Log.Info("Recent batcher txs detected. Need to wait for more block confirms", "confirmsNeeded", blockConfirms)
+	// Decrease block num until we find the block before the most recent batcher tx was sent
+	for currentNonce != previousNonce {
+		currentBlock.Sub(currentBlock, big.NewInt(1))
+		currentNonce, err = l.L1Client.NonceAt(l.shutdownCtx, l.Txmgr.From(), currentBlock)
+		if err != nil {
+			l.Log.Error("Failed to retrieve nonce", "err", err)
+			return
+		}
+	}
+
+	batcherTxFinalizedBlock := currentBlock.Uint64() + 1 + blockConfirms
+	err = dial.WaitRollupSync(l.shutdownCtx, l.Log, rollupClient, batcherTxFinalizedBlock, time.Second*5)
```

Comment:
```suggestion
	batcherTxBlock := currentBlock.Uint64() + 1
	err = dial.WaitRollupSync(l.shutdownCtx, l.Log, rollupClient, batcherTxBlock, time.Second*12)
```
We don't need to handle the verifier confirmation depth, because `WaitRollupSync` waits for the `CurrentL1` block to catch up, which already is `VerifierConfDepth` deep.

Since we're waiting on L1 blocks, it makes sense to poll every 12 sec.